sdk/servicebus/service-bus/src/serviceBusClient.ts

```diff
@@ -227,8 +227,7 @@ export class ServiceBusClient {
       `${entityPath}/${generate_uuid()}`
     );
 
-    // TODO: .NET actually tries to open the session here so we'd need to be async for that.
-    return new SessionReceiverImpl(
+    const sessionReceiverImpl = SessionReceiverImpl.createInitializedSessionReceiver(
```

Comment:
I would recommend that we either add an await here or directly return.
One would expect that a const named `sessionReceiverImpl` to have the actual implementation, but here it has the promise

Reply:
Ah, meant to remove that. :)